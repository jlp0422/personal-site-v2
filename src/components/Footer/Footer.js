import React from "react"

const Footer = () => (
  <footer>
    &copy; {new Date().getFullYear()} Jeremy L. Philipson
  </footer>
)

export default Footer
