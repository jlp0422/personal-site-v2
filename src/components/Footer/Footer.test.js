import React from 'react'
import renderer from 'react-test-renderer'
import Footer from './Footer'

describe('Footer', () => {
  const getOutput = () => renderer.create(<Footer />)

  it('Renders correctly', () => {
    const tree = getOutput().toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Displays the copyright, current year, and full name', () => {
    const instance = getOutput().toJSON()
    expect(instance.type).toBe('footer')
    expect(instance.props).toEqual({})
    expect(instance.children.map(str => str.trim())).toEqual([
      'Â©',
      '2021',
      'Jeremy L. Philipson'
    ])
  })
})
