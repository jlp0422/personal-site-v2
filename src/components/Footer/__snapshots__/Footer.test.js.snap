// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Renders correctly 1`] = `
<footer>
  Â© 
  2020
   Jeremy L. Philipson
</footer>
`;
